plugins/woocommerce/includes/wc-product-functions.php

```
@@ -769,8 +769,40 @@ function wc_maybe_schedule_product_sale_events( $product_id, $product = null ):
+ *
+ * Hooks into post meta operations so per-product sale events are kept in sync regardless
+ * of how the meta is written: WooCommerce CRUD, direct update_post_meta() calls from
+ * importers, ERP sync tools, or custom code.
+ *
+ * @since 10.8.0
+ * @param int|int[] $meta_id    Meta ID (or array of IDs for delete).
+ * @param int       $object_id  Post ID.
+ * @param string    $meta_key   Meta key.
+ * @return void
+ */
+function wc_maybe_schedule_sale_events_on_meta_change( $meta_id, $object_id, $meta_key ): void { // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter.FoundAfterLastUsed
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$meta_id'. (undefined)

(UnusedFormalParameter)